bigbluebutton-html5/imports/ui/components/audio/service.js

```diff
@@ -34,14 +37,32 @@ const init = (messages, intl) => {
 
 const currentUser = () => mapUser(Users.findOne({ intId: Auth.userID }));
 
+const toggleMuteMicrophone = () => {
+  makeCall('toggleSelfVoice');
+  const cvu = VoiceUsers.findOne({ meetingId: Auth.meetingID, intId: Auth.userID });
```

Comment:
This is prone to race conditions. A way to avoid it (albeit redundant) would be:


```
const toggleMuteMicrophone = () => {
  const cvu = VoiceUsers.findOne({ meetingId: Auth.meetingID, intId: Auth.userID }, { fields: { muted: 1 } });
  if (cvu) {
    if (cvu.muted) {
      logger.info({
        logCode: 'audiomanager_unmute_audio',
        extraInfo: { logType: 'user_action' },
      }, 'microphone unmuted by user');
      makeCall('toggleSelfVoice');
    } else {
      logger.info({
        logCode: 'audiomanager_mute_audio',
        extraInfo: { logType: 'user_action' },
      }, 'microphone muted by user');
      makeCall('toggleSelfVoice');
    }
  }
};
```

Notice the use of `fields` limiter -- it will cause the query to return only some pieces of the entire 'document'